build.sbt

```diff
@@ -113,7 +113,11 @@ lazy val commonSettings = Seq(
     "-language:experimental.macros",
     "-unchecked",
     "-Xfatal-warnings",
-    "-Xxml:-coalescing",
+    "-Xxml:-coalescing"
+  ),
+  javacOptions ++= Seq(
+    "-Werror",
+    "-Xlint:deprecation"
   ),
```

Comment:
I noticed an issue when running ``sbt universal:packageBin``. This command calls ``sbt doc`` for all subprojects. In particular, it calls ``sbt daffodil-udf/doc``. It looks like because there is Java code in this repo is calls javadoc, and these javacOptions get passed to it. But these aren't valid options for javadoc, only for javac. There's an sbt issue talking about this behavior: https://github.com/sbt/sbt/issues/355

Looks like the solution is to do ``javacOptions in Compile in compile ++= ...``

Reply:
Fixed